Sanitizers.cmake

```diff
@@ -0,0 +1,76 @@
+# Taken from https://github.com/cpp-best-practices/cpp_starter_project
+# Originally by Jason Turner/lefticus, released under the "unlicense" license
+# into the public domain
+
+# Adds build options to turn on the various sanitizers to a target, all turned off by default.
+# - ENABLE_SANITIZER_ADDRESS
+# - ENABLE_SANITIZER_LEAK
+# - ENABLE_SANITIZER_UNDEFINED_BEHAVIOR
+# - ENABLE_SANITIZER_THREAD
+# - ENABLE_SANITIZER_MEMORY
+#
+# each of which turns on the corresponding -fsanitize= flag for the project_name
+# argument, both as compilation and
+#
+# Note that some options are incompatible, which will produce warnings.
+function(enable_sanitizers project_name)
```

Comment:
Is there a way to check for the sanitizers within the CMake function `enable_sanitizers`? I just want to add a simple message to help the users debug if the santizers are not working. Something like

```cmake
if (NOT sanitizers_FOUND)
  message(WARN "Santizers were not found in this environment. Have you upgraded to at least version 3.2 of the development container? 'ldmx pull dev v3.2'.")
endif()
```
If the user tries to `ENABLE` some when using an older dev container.